### ISSUE TITLE
DueDate reported with wrong timezone

### ISSUE THREAD

**Issue by pbek** [nextcloud/deck#4764]
I don't know if this exactly the rather old issue #2276.
But I still have the same problem on Nextcloud 26.0.2 (docker) with `TZ=Europe/Vienna`.

I'm working on a feature for creating Deck Cards in QOwnNotes (https://github.com/pbek/QOwnNotes/issues/2789).

When I use the [Deck API](https://deck.readthedocs.io/en/latest/API/) to create a Card, I send:

```json
{"duedate":"2023-06-06T06:15:30+02:00","order":0,"title":"tz4","type":"plain"}
```

I get back:

```json
{"ETag":"8ef788d31fae586d508563b5b52c5de1","archived":false,"assignedUsers":null,"attachmentCount":null,"attachments":null,"commentsCount":0,"commentsUnread":0,"createdAt":1686024933,"deletedAt":0,"description":"","descriptionPrev":null,"duedate":"2023-06-06T06:15:30+02:00","id":78,"labels":null,"lastEditor":null,"lastModified":1686024933,"notified":false,"order":0,"owner":"test","relatedBoard":null,"relatedStack":null,"stackId":16,"title":"tz4","type":"plain"}
```

Note that the timezones are both correct, `2023-06-06T06:15:30+02:00`.
But when I look at the Deck page on my Nextcloud server, the timezone is not correct at all!

`2023-06-06T06:15:30+00:00`

![image](https://github.com/nextcloud/deck/assets/1798101/7655c89b-9970-4111-8fbb-ed105a0127a4)

I can't see any useful Nextcloud Log errors/warning in that period.

**Comment by pbek** [comment]
Timezone looks correct in mariadb too...

![image](https://github.com/nextcloud/deck/assets/1798101/be60b98d-e6bb-41ea-9028-08c3c18b205d)

**Comment by pbek** [comment]
No, the timezone in the DB is not correct, the MySQL dump showed me `'2023-06-06 06:15:30'`, no TZ, so it's stored without TZ (= UTC).

**Comment by pbek** [comment]
When I create an entry in the Web UI with 20:00 it's stored correctly as `'2023-06-06 18:00:00'` in the DB. So Deck seems to store DateTimes without converting it first to UTC via the public API.

**Comment by pbek** [comment]
I don't know if it's the same issue, but for example [here](https://github.com/digital-blueprint/relay-dispatch-bundle/blob/4d488e5cb56cc9ea9223591aee558eacf01a1074/src/Service/DispatchService.php#L483https://github.com/digital-blueprint/relay-dispatch-bundle/blob/4d488e5cb56cc9ea9223591aee558eacf01a1074/src/Service/DispatchService.php#L483) I had to convert `new \DateTimeImmutable('now', new DateTimeZone('UTC')` to force the conversion. 

[Here](https://github.com/digital-blueprint/dbp-relay-dispatch-bundle/blob/4d488e5cb56cc9ea9223591aee558eacf01a1074/src/Entity/Request.php#L7) I even had to force `date_default_timezone_set('UTC');` to get the API to output UTC.

**Comment by pbek** [comment]
I found https://github.com/nextcloud/deck/blob/8038e568a3a8ecec29fa8747a2981426d62f3b3d/lib/Service/CardService.php#L238, but I guess at that time there are still strings...

Maybe dealing with it in https://github.com/nextcloud/deck/blob/8038e568a3a8ecec29fa8747a2981426d62f3b3d/lib/Db/CardMapper.php#L77, but I'm fishing in the dark, since Nextcloud seems to handle those conversion.

**Comment by Fufs** [comment]
Hi,
I stumbled on this issue, while looking for a solution for a different one. I think the problem here is that `$duedate` is passed directly when creating a new card. This is bad because the MySQL datetime type is not timezone aware, as per this [doc](https://dev.mysql.com/doc/refman/8.0/en/datetime.html#:~:text=The%20DATETIME%20type%20is%20used%20for%20values%20that%20contain%20both%20date%20and%20time%20parts.%20MySQL%20retrieves%20and%20displays%20DATETIME%20values%20in%20%27YYYY%2DMM%2DDD%20hh%3Amm%3Ass%27%20format.%20The%20supported%20range%20is%20%271000%2D01%2D01%2000%3A00%3A00%27%20to%20%279999%2D12%2D31%2023%3A59%3A59%27.). 

I think someone has encountered this problem before, because if you scroll a bit down, you can see this issue is patched for the update function

https://github.com/nextcloud/deck/blob/bfe2a9053f4d54d5c6b64f6822fabf5143a555ac/lib/Service/CardService.php#L340C1-L340C64

The solution then is to change the create function from
```php
$card->setDuedate($duedate); 
```
to
```php
$card->setDuedate($duedate ? new \DateTime($duedate) : null);
```

Hope that helps!
